third_party/move/move-model/src/well_known.rs

```diff
@@ -45,15 +45,21 @@ pub const TYPE_NAME_GET_SPEC: &str = "type_name::$get";
 /// The well-known name of the first parameter of a method.
 pub const RECEIVER_PARAM_NAME: &str = "self";
 
-/// The well-known abort codes used by the compiler. These are marked
-/// by lowest 6 bytes of a sha256 of the string "Move 2 Abort Code",
-/// appended with two bytes for the error type.
-/// TODO: add a check at runtime that user is not clashing with reserved
-/// codes?
-pub const WELL_KNOWN_ABORT_CODE_BASE: u64 = 0xD8CA26CBD9BE << 16;
+/// The well-known abort codes used by the compiler. Those confirm
+/// to the error category standard as defined in `errors.move`
+/// in the standard library. The lowest three bytes represent
+/// the error category (one byte) and the reason (two bytes).
+/// All compiler generated abort cords use category
+/// `std::error::INTERNAL` (`0xB`). The upper five bytes
+/// are populated with the lowest bytes of the sha256
+/// of the string "Move 2 Abort Code".
+const fn make_abort_code(n: u64) -> u64 {
+    let magic = 0xCA26CBD9BE; // sha256("Move 2 Abort code")
+    (magic << 24) | 0xB << 16 | n
+}
 
 // Used when user omits an abort code.
-pub const UNSPECIFIED_ABORT_CODE: u64 = WELL_KNOWN_ABORT_CODE_BASE;
+pub const UNSPECIFIED_ABORT_CODE: u64 = make_abort_code(0);
 
 // Used when a runtime value falls through a match.
-pub const INCOMPLETE_MATCH_ABORT_CODE: u64 = WELL_KNOWN_ABORT_CODE_BASE | 0x1;
+pub const INCOMPLETE_MATCH_ABORT_CODE: u64 = make_abort_code(1);
```

Comment:
I remember seeing the abort codes mentioned in the Move book. We should update them to whatever these new ones are.

Reply:
Did already.